src/custom-element.js

```diff
@@ -767,8 +767,16 @@ function createBaseAmpElementProto(win) {
     if (this.isInTemplate_) {
       return;
     }
-    this.resources_.remove(this);
-    this.implementation_.detachedCallback();
+    try {
+      this.implementation_.viewportCallback(false);
+    } finally {
+      try {
+        this.implementation_.unlayoutCallback();
```

Comment:
One question: do we have to call `unlayoutCallback()` for this.element owned elements?